src/ViewLayer/RaceModeDisplay/RaceModeDashboardView.cpp

```diff
@@ -187,15 +189,89 @@ void RaceModeDashboardView::initalizeFaultAnimation()
 
 void RaceModeDashboardView::runFaultAnimation()
 {
-    int numberOfActiveFaults = batteryFaultsList_.numberOfActiveFaults() +
-                               motorOneFaultsList_.numberOfActiveFaults() +
-                               motorZeroFaultsList_.numberOfActiveFaults();
+    QVector<QString> batteryFaults = batteryFaultsList_.activeFaultLabels();
+    QVector<QString> motorZeroFaults = motorZeroFaultsList_.activeFaultLabels();
+    QVector<QString> motorOneFaults = motorOneFaultsList_.activeFaultLabels();
 
-    if (numberOfActiveFaults > 0 && faultAnimation_->state() != QAbstractAnimation::Running)
+    if (faultAnimation_->state() != QAbstractAnimation::Running && triggerFaultAnimation(batteryFaults, motorZeroFaults, motorOneFaults))
     {
         faultAnimation_->setDirection(QAbstractAnimation::Forward);
         faultAnimation_->start();
     }
+
+    prevBatteryFaults_ = batteryFaults;
+    prevMotorZeroFaults_ = motorZeroFaults;
+    prevMotorOneFaults_ = motorOneFaults;
+}
+
+bool RaceModeDashboardView::triggerFaultAnimation(QVector<QString>& batteryFaults, QVector<QString>& motorZeroFaults, QVector<QString>& motorOneFaults)
```

Comment:
I think triggerFaultAnimation is a confusing name. It seems more like an action, rather than what its actually doing, which is just checking if we should run the fault animation.

If you agree, then I suggest changing the name (though now that I'm thinking about what to rename the function to, its kinda hard..... - maybe something like faultAnimationCheck?).

Reply:
Also, this contains a lot of repeated code.

Can this be changed to
`bool RaceModeDashboardView::faultAnimationCheck(QVector<QString> currentFaults, QVector<QString> prevFaults)`

and we would call it like this
`
if (
  faultAnimation_->state() != QAbstractAnimation::Running && 
  faultAnimationCheck(batteryFaults, prevBatteryFaults_) &&
  faultAnimationCheck(motorZeroFaults, prevMotorZeroFaults_) &&
  faultAnimationCheck(motorOneFaults, prevMotorOneFaults_)
)
`

(FYI, I know I'm being a hypocrite, since I said I don't like having lots of conditions in another comment, so forgive me)

Reply:
No no, what you said makes sense. I've incorporated both these comments.